src/he/he_internal.h

```diff
@@ -263,12 +268,20 @@ typedef struct he_msg_hdr {
 
 typedef struct he_msg_ping {
   he_msg_hdr_t msg_header;
-  uint32_t payload;
+  /// Identifier for matching the reply message
```

Comment:
Thinking out loud, please check my understanding: 

The intention here is to be backwards compatible with older versions of LIghtway that are not sending ping IDs right?  Prior to this change, old versions of Lightway would have already initialized `uint32_t payload` to 0.
The new version of Pong will echo back the first half of this 0 value as Pong ID, and then the older version of Lightway that sent the Ping will just safely ignore this payload of 0 Pong ID.

Do we need to reserve Ping ID of 0 as a special hint that it's an older version of Lightway on the other side?  WIth that hint, perhaps we should explicitly ignore the `length` (even though it'll likely be 0) and `payload`? (Maybe we should only cross that bridge when we need to start interpreting the `length` field)

Reply:
> Do we need to reserve Ping ID of 0 as a special hint that it's an older version of Lightway on the other side?

I don't think it's necessary, because we'll ship the server with this new lightway-core first, then it should be all backward compatible:

* Old client -> new servers: the payload field in old struct is always 0, the server will see the id as 0, and echo it back. Old client doesn't check the payload in pong message.
* New client -> new server: the id field will be set and server will echo it back as designed.

Reply:
Edited PR description to clarify it.

Reply:
I meant we might have a case with "old client -> new server", and new server trying to interpret the length of 0 as something meaningful. I agree that it's not necessary to worry about that in this PR. It might not be an issue in the future depending on how future PRs go.

Reply:
> I meant we might have a case with "old client -> new server", and new server trying to interpret the length of 0 as something meaningful.

Got it. No need to worry about it even in the future. The design is, the receiver of the `ping` message should ignore the payload.